### PR TITLE
fix(metrics): skip empty string dimension values

### PR DESCRIPTION
**Pull request by arnabrahman**
## Summary
When adding a dimension to my metrics, I should get a warning if the dimension value is invalid aka an empty string ("") or undefined/null, and the dimension should not be added to the EMF blobs emitted by the utility.

### Changes

- Check for `empty`, `undefined` & `null` values for dimension and if found log warning
- Skip invalid value dimensions to EMF, do this for `addDimension` & `addDimensions`
- Unit test the fix

**Issue number:** #3302 

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Thank you for the revisions @arnabrahman!

The friends at the Python version told me they already merged the change and used a slightly different wording in the warning log, which you [can see here](https://github.com/aws-powertools/powertools-lambda-python/commit/f8fe2a3c60e132cf211090417b83eb6ff9dffbc2#diff-d4a359e5caba54c548936f54ad216c2b6a81f00929e3dc2b33cba5b43c6b84caR274-R279) and is this:

> The dimension {name} doesn't meet the requirements and won't be added. Ensure the dimension name and value are non empty strings"

As last step before merging, could you please change to this?

I'll merge right after, sorry I missed the change on their part and didn't point it out in my first review of your PR.

**Comment by arnabrahman** [comment]
It's ok, i have updated the warning message.

**Review by dreamorosi — APPROVED** [review]
Thanks for addressing the feedback, let's merge it!

**Pull request merged**
